b2luigi/batch/processes/gbasf2.py

```diff
@@ -624,7 +624,7 @@ def _failed_files_from_dataset_download(self, stdout):
         failed_files = stdout.split('Failed files:')[-1].\
             split("Files with duplicated jobID, not downloaded:")[0].\
             split("Skip ")[0].strip().split('\n')
-        failed_files = [line for line in failed_files if len(line.strip()) > 0]
+        failed_files = [line.split(' to ')[0].strip() + '.root' for line in failed_files if len(line.strip()) > 0]
```

Comment:
1. Instead of `if len(line.strip()) > 0`, maybe we should check that the new substring before the `' to '` is not empty, that is 
``` python
if len(line.split(' to ')[0].strip()) > 0
```
2. I just noticed that probably we can use the implicity booleaness of strings for a more pythonic check, together with the suggestion from 1. that would be 
``` python
if line.split(' to ')[0].strip()
```
3. I can imagine that in a future release the `' to '` output might change to another string (e.g. to an arrow), but I expect that the file names will always be root files. Therefore, instead of splitting on ` to ` and appending `.root`, I think it would be more future safe to select the text in each line upto including `.root`.  For that on can either use a regex with the `re` package or just split on `.root` and adding it again.

Maybe I'll add some suggestions tomorrow that you can just accept.

Reply:
@meliache I'll add the suggestions 1 and 2 directly, I think doing it in what way looks better.

Concerning suggestion 3: yeah this seems still to be bugged the adding of `.root` should not be there in case of using ` to ` for splitting. The reason why it is still there is because I had splitting by `.root` in the commits before but thought, that might be too specific for output file extension. However, if you think that covers most of the application cases, I will introduce the splitting by `.root` back

Reply:
@meliache  I've had an example with a failed file. Seems like there is a bug on `gb2_ds_get` side, that the option `--input_dslist` corrupt and `No file found` is printed out. I'll have a look at that.

Reply:
So all in all, my fix won't work although now better catching the case, but it is a problem on `gbasf2` side. I'll have a look at the code and will report here the fix.

Reply:
Found the issue...

https://stash.desy.de/projects/B2DC/repos/belledirac/browse/Client/controllers/datasetCLController.py#142

`line[:-1]` should be replaced with `line.strip()`...

I've created a JIRA Ticket for that https://agira.desy.de/projects/BIIDCD/issues/BIIDCD-1408